wordpress-mobile/WordPress-Android
# Prepublishing Nudges: PostSettingsTagsActivity extracted to PostSettingsTagsFragment

Fixes #11643

## Findings

The goal of this PR is to extract the select tags functionality being used in the `Post Settings`. 
Based on my analysis the functionality that exists within the current `PostSettingsTagsActivity` is identical to what is needed for this project. The only difference seems to be the header that is above the `EditText` control based on the iOS design, the navigation and the bottom sheet that is housing it. 

Prepublishing Nudges IOS - Tags | Post Settings Android - Tags 
--------|-------
<img src="https://user-images.githubusercontent.com/1509205/79089792-0d55e000-7d0d-11ea-8e6c-fdf79544ef20.png" width="320">        | <img src="https://user-images.githubusercontent.com/1509205/79089843-38d8ca80-7d0d-11ea-9fb7-68fc5d6c71da.png" width="320">

## Solution
A new fragment was created called `PostSettingsTagsFragment` which hosts pretty much all the code that existed within `PostSettingsTagsActivity`. The main differences are as follows: 

1. The fragment doesn't contain the toolbar since both screens that will use the fragment will handle their navigational toolbar independently. 

2. There is now a listener that can be implemented by the parent `PostSettingsTagsActivity` to get updates when the tags within the fragment get changed. 

3. The keyboard closing is being triggered by the parent activity that finds the fragment by tag and calls the `closeKeyboard` method. 

When thinking to do this, I decided to simply move the code to a fragment that uses Java  (instead of converting to Kotlin) which would allow for any historical comments, changes etc. to remain. Let me know if this decision is fine or if I should be converting this fragment since we are touching it. 
I also didn't modify the existing code or logic in any way. There are optimizations/cleanups that could be done but since that is outside the scope of this project, I simply reused the logic as is. My main reason for taking this approach is that I think it will make maintaining this functionality a lot easier in the future as both components that utilize it, will be able to benefit from the shared logic. 

## Testing
1. Create a new post. 
2. Click the options menu. 
3. Click Post settings. 
4. Click Tags. 
5. Add some tags. ( Ensure that existing tags are there so the list functionality can be observed.) 
6. Press back and ensure the tags that were selected are now visible. 
7. You can save a draft and reopen the post to ensure the tags exist.

## Reviewing 
1. Only 1 reviewer is needed but anyone can review. 
2. Ensure that [Prepublishing Nudges : create core bottom sheet functionality](https://github.com/wordpress-mobile/WordPress-Android/pull/11626) is merged. 
3. Set this base branch to `feature/prepublishing_nudges`
4. Remove `Not Ready for Merge`
5. Merge :) 

## Submitter Checklist
